File: core/src/main/java/com/sequenceiq/cloudbreak/service/BuildStackFailureException.java
package com.sequenceiq.cloudbreak.service;

import java.util.HashSet;
import java.util.Set;

import com.sequenceiq.cloudbreak.domain.Resource;

public class BuildStackFailureException extends CloudbreakServiceException {

    private final Set<Resource> resourceSet;

    public BuildStackFailureException(Exception ex) {
        super(ex);
        this.resourceSet = new HashSet<>();
    }

    public BuildStackFailureException(String message, Throwable cause) {
        super(message, cause);
        this.resourceSet = new HashSet<>();
    }

    public BuildStackFailureException(String message, Throwable cause, Set<Resource> resourceSet) {
        super(message, cause);
        this.resourceSet = resourceSet;
    }

    public Set<Resource> getResourceSet() {
        return resourceSet;
    }
}

File: core/src/main/java/com/sequenceiq/cloudbreak/service/BuildStackFailureException.kt
package com.sequenceiq.cloudbreak.service

import java.util.HashSet

import com.sequenceiq.cloudbreak.domain.Resource

class BuildStackFailureException : CloudbreakServiceException {

    val resourceSet: Set<Resource>

    constructor(ex: Exception) : super(ex) {
        this.resourceSet = HashSet<Resource>()
    }

    constructor(message: String, cause: Throwable) : super(message, cause) {
        this.resourceSet = HashSet<Resource>()
    }

    constructor(message: String, cause: Throwable, resourceSet: Set<Resource>) : super(message, cause) {
        this.resourceSet = resourceSet
    }
}
